src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/ICodeActionProvider.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    internal interface ICodeActionProvider
+    {
+        Task<IReadOnlyList<RazorVSInternalCodeAction>?> ProvideAsync(RazorCodeActionContext context, IEnumerable<RazorVSInternalCodeAction> codeActions, CancellationToken cancellationToken);
```

Comment:
I think this could generally use some DocComments. For example it's not clear to me why a `ICodeActionProvider` would take in CodeActions. Is this more like `ICodeActionResolver`? If so what does it mean to return null from the resolver? I would have assumed that failing to resolve would just return the unresolved actions.

Reply:
Yeah, I agree. I was originally planning on renaming the "provider" types, because they are confusing, but I couldn't come up with a better name. Will doc.